src/transformers/generation_utils.py

```diff
@@ -39,6 +40,69 @@
 logger = logging.get_logger(__name__)
 
 
+@dataclass
+class GreedySearchDecoderOnlyOutput(ModelOutput):
+    """
+    Base class for outputs of encoder-decoder generation models using greedy search.
+
+    Args:
+        sequences (:obj: `torch.LongTensor` of shape :obj:`(batch_size * num_return_sequences, sequence_length)`):
+            The generated sequences. The second dimension (sequence_length) is either equal to :obj:`max_length` or shorter if all
+            batches finished early due to the :obj:`eos_token_id`.
+        logits (:obj:`torch.FloatTensor` of shape :ob:`(batch_size * num_return_sequence, sequence_length, config.vocab_size)`, `optional`, returned when ``output_scores=True`` is passed or when ``config.output_scores=True``):
+            Classification scores for each token of the generated sequences. :obj:`(max_length,)`-shaped
+            tuple of :obj:`torch.FloatTensor` of shape :obj:`(batch_size * num_return_sequence, config.num_labels)`.
+        attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer of the decoder) of shape
```

Comment:
will be tuple of tuple (memory explosion :D) we need attentions and hidden_states of **all** layers at **each** generation iter